src/Vector.ts

```diff
@@ -150,18 +151,27 @@ export default class Vector {
 
     Dimension.checkDimensions(v1.dimensions, v2.dimensions)
 
-    const entries = _.chain(v1.entries.concat(v2.entries))
-      .groupBy((entry: VectorEntry) => entry.coord.toString())
-      .values()
-      .map((grouped: VectorEntry[]) => {
-        const coord = [...grouped[0].coord]
-        const value = grouped.map((entry) => entry.value).reduce((a, b) => a.add(b))
-        return new VectorEntry(coord, value)
-      })
-      .filter((entry) => !entry.value.isAlmostZero())
-      .value()
+    // this function is very hot, so loops are hand-rolled
+    const entriesByCoord: Record<string, VectorEntry> = {}
+    for (const entry of v1.entries) {
+      const key = entry.coord.toString()
+      entriesByCoord[key] = entry
+    }
+    for (const entry of v2.entries) {
+      const key = entry.coord.toString()
+      if (entriesByCoord.hasOwnProperty(key)) {
+        const value = entriesByCoord[key].value.add(entry.value)
+        if (value.isAlmostZero()) {
+          delete entriesByCoord[key]
+        } else {
+          entriesByCoord[key] = new VectorEntry(entry.coord, value)
+        }
+      } else {
+        entriesByCoord[key] = entry
+      }
+    }
```

Comment:
Same as before: would really prefer to save a shorter implementation as a reference.